### PR TITLE
Added Traffic split annotations

### PR DESCRIPTION
**Pull request by Vidya2606**

### CODE REVIEW

File: src/strategyHelpers/blueGreen/smiBlueGreenHelper.ts [hunk 1]
```diff
@@ -127,7 +127,8 @@ let trafficSplitAPIVersion = ''
 async function createTrafficSplitObject(
    kubectl: Kubectl,
    name: string,
-   nextLabel: string
+   nextLabel: string,
+   annotations: {[key: string]: string} = {}
```
Comment (OliverMKing) [inline src/strategyHelpers/blueGreen/smiBlueGreenHelper.ts:131]: how are you letting a user pass this in? You need to add logic to let a user insert it.

We also need to do the same thing for canary.

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: src/run.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export async function run() {
    const action: Action | undefined = parseAction(
       core.getInput('action', {required: true})
    )
+   const annotations: {[key: string]: string} = {}
```
Comment (OliverMKing) [inline src/run.ts:21]: you aren't passing in the annotations with core.getInput

File: action.yml [hunk 3]
```diff
@@ -35,6 +35,10 @@ inputs:
       description: 'Traffic split method to be used. Allowed values are pod and smi'
       required: false
       default: 'pod'
+   traffic-split-annotations:
+      description: 'Traffic split annotations in the form of key/value pair can be used'
+      required: false
+      default: 0
```
Comment (OliverMKing) [inline action.yml:41]: why a default of 0?

File: src/run.ts [hunk 4]
```diff
@@ -38,11 +39,11 @@ export async function run() {
          break
```
Comment (OliverMKing) [inline src/run.ts:39]: you likely need to add this to the deploy scenario as well

File: src/run.ts (hunk 2 above)
Comment (OliverMKing) [reply src/run.ts:21]: also keep in mind that with core.getInput you will be parsing it as a string. You will have to parse it as an object

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: action.yml [hunk 5]
```diff
@@ -35,6 +35,9 @@ inputs:
       description: 'Traffic split method to be used. Allowed values are pod and smi'
       required: false
       default: 'pod'
+   traffic-split-annotations:
+      description: 'Traffic split annotations in the form of key/value pair can be used'
```
[the remark this answers is not in the document]
Comment (OliverMKing) [reply action.yml:39]: Also mention that it's relevant only if strategy == canary or blue green

**Review by OliverMKing — CHANGES_REQUESTED** [review]
Looking good! Small doc changes then we should be good.

File: src/strategyHelpers/deploymentHelper.ts [hunk 6]
```diff
@@ -41,7 +41,8 @@ export async function deployManifests(
    files: string[],
    deploymentStrategy: DeploymentStrategy,
    kubectl: Kubectl,
-   trafficSplitMethod: TrafficSplitMethod
+   trafficSplitMethod: TrafficSplitMethod,
+   annotaions: {[key: string]: string} = {}
```
Comment (davidgamero) [inline src/strategyHelpers/deploymentHelper.ts:45]: nit: typo annotaions -> annotations

File: src/strategyHelpers/blueGreen/smiBlueGreenHelper.ts [hunk 7]
```diff
@@ -91,7 +94,11 @@ export async function rejectBlueGreenSMI(
    await cleanupSMI(kubectl, manifestObjects.serviceEntityList)
 }
 
-export async function setupSMI(kubectl: Kubectl, serviceEntityList: any[]) {
+export async function setupSMI(
+   kubectl: Kubectl,
+   serviceEntityList: any[],
+   annotaions: {[key: string]: string} = {}
```
Comment (davidgamero) [inline src/strategyHelpers/blueGreen/smiBlueGreenHelper.ts:100]: nit: typo

**Review by OliverMKing — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]

**Pull request merged**